### DISCUSSION TITLE
How Hypha handle the partner feature

### DISCUSSION THREAD

**Discussion by frjo** [OpenTechFund/opentech.fund#4016] [General]
Related https://github.com/HyphaApp/hypha/pull/3913 and https://github.com/HyphaApp/hypha/pull/3893

The resent PRs to fix partner permissions etc. work well and solves issues that need solving.

But to me the solutions look a bit "messy" and non obvious.

I think OTF uses partners in a quite different manner from how the feature was built initially. And now we push and pull on it to make it do what OTF wants today.

I think the partner feature would work a lot better if we redesigned it from the ground up to conform to OTFs needs.

The UI and naming can stay the same so as not to confuse the users but we rebuilt/rethink the backend.

It is not a lot of code I think, more work with deciding what to code than to code it. The result would be a feature that does exactly what OTF needs and that are clear and easy to work with for future developers.

**Comment by wes-otf** [comment]
I'll also bring this up in the maintainers meeting tomorrow to see where their vision/usage of the partner role fits in this

**Comment by wes-otf** [comment]
Closed by #4790
